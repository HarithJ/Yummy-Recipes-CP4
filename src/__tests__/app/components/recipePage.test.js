import React from 'react';
import { shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson, { shallowToJson } from 'enzyme-to-json';

import RecipePage from '../../../components/recipes/recipePage.js'

describe("recipe page component", () => {
  const event = {
    target: { value: '' },
    preventDefault: jest.fn(),
  };

  it("renders properly", () => {
    const recipePage = shallow(<RecipePage match={{params: {}}} />)

    expect(shallowToJson(recipePage)).toMatchSnapshot();
  })

  it("has a function to set search value", () => {
    const recipePage = shallow(<RecipePage match={{params: {}}} />);

    recipePage.setProps({"history": {"push": jest.fn()}})

    expect(recipePage.instance().setSearchValue(event));
  })

  it("has a function to handle changing of a page", () => {
    const recipePage = shallow(<RecipePage match={{params: {}}} />);

    expect(recipePage.instance().changePage(3));
  })

  it("has a function to handle changing recipe limits", () => {
    const recipePage = shallow(<RecipePage match={{params: {}}} />);

    expect(recipePage.instance().changeLimit(3));
  })

  it("has a function to set total number of pages", () => {
    const recipePage = shallow(<RecipePage match={{params: {}}} />);

    expect(recipePage.instance().setTotalPageNumbers());
  })

  it("has a function to set alert", () => {
    const recipePage = shallow(<RecipePage match={{params: {}}} />);

    expect(recipePage.instance().setAlertMsg());
  })

  it("has a function to hide alert", () => {
    const recipePage = shallow(<RecipePage match={{params: {}}} />);

    expect(recipePage.instance().hideAlert());
  })
})
