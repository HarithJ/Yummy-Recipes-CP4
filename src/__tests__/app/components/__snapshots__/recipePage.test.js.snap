// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipe page component renders properly 1`] = `
<div>
  <Nav
    setSearchValue={[Function]}
  />
  <div
    class="container-fluid mt-3"
  >
    <Recipes
      pagination={
        Object {
          "limit": 0,
          "page": 1,
          "totalPages": 1,
        }
      }
      searchValue=""
      setAlertMsg={[Function]}
    />
    <Pagination
      changeLimit={[Function]}
      changePage={[Function]}
      pagination={
        Object {
          "limit": 0,
          "page": 1,
          "totalPages": 1,
        }
      }
    />
    <Link
      replace={false}
      to="/categories"
    >
      <i
        class="btn-success fas fa-arrow-alt-circle-left"
        style={
          Object {
            "fontSize": "2.5em",
            "position": "fixed",
            "top": "50%",
          }
        }
      />
    </Link>
  </div>
</div>
`;
